joschahenningsen/TUM-Live
# Improvement/refactor course sections

### Motivation and Context

https://github.com/TUM-Dev/gocast/pull/1137 made the code shorter and brought it in a better and reactive structure. The videosections have been following an own update cycle. It would be nice to have one united update cycle experience for a lecture. Giving the user the opportunity to discard or save all changed data by one button.

### Description
Integrated the course section editing to the lecture edit section.

### Steps for Testing

- Go to course management Page
- Open a course
- Edit a lecture
- Add / Delete / Change Video Sections
- Test: Discard & Save
- Reload and check if data is persistent

### Screenshots

Super reactive UX with `change-set` & `DataProviders`

https://github.com/TUM-Dev/gocast/assets/17506411/782f97cf-ea3a-44af-98d0-195ce29422e6

Complex reactive validation

https://github.com/TUM-Dev/gocast/assets/17506411/8111cfe0-6cd6-427a-93d3-b79f8e35cec6


also fixes #1178
